drivers/stm32/canard_stm32.h

```diff
@@ -0,0 +1,335 @@
+ *
+ * This function can be invoked any number of times; every invocation re-initializes everything from scratch.
+ *
+ * WARNING: The clock of the CAN module must be enabled before this function is invoked!
+ *          If CAN2 is used, CAN1 must be also enabled!
+ *
+ * WARNING: The driver is not thread-safe!
+ *          It does not use IRQ or critical sections though, so it is safe to invoke its API functions from the
+ *          IRQ context from the application.
+ *
+ * @retval      0               Success
+ * @retval      negative        Error
+ */
+int canardSTM32Init(const CanardSTM32CANTimings* const timings,
+                    const CanardSTM32IfaceMode iface_mode);
+
+/**
+ * Pushes one frame into the TX buffer, if there is space.
+ * Note that proper care is taken to ensure that no inner priority inversion is taking place.
+ * This function does never block.
+ *
+ * @retval      1               Transmitted successfully
+ * @retval      0               No space in the buffer
+ * @retval      negative        Error
+ */
+int canardSTM32Transmit(const CanardCANFrame* const frame);
+
+/**
+ * Reads one frame from the hardware RX FIFO, unless all FIFO are empty.
+ * This function does never block.
+ *
+ * @retval      1               Read successfully
+ * @retval      0               The buffer is empty
+ * @retval      negative        Error
+ */
+int canardSTM32Receive(CanardCANFrame* const out_frame);
+
+/**
+ * Sets up acceptance filters according to the provided list of ID and masks.
+ * Note that when the interface is reinitialized, hardware acceptance filters are reset.
+ * Also note that during filter reconfiguration, some RX frames may be lost.
+ *
+ * Setting zero filters will result in rejection of all frames.
+ * In order to accept all frames, set one filter with ID = Mask = 0, which is also the default configuration.
+ *
+ * @retval      0               Success
+ * @retval      negative        Error
+ */
+int canardSTM32ConfigureAcceptanceFilters(const CanardSTM32AcceptanceFilterConfiguration* const filter_configs,
+                                          const unsigned num_filter_configs);
+
+/**
+ * Returns the runnning interface statistics.
+ */
+CanardSTM32Stats canardSTM32GetStats(void);
+
+/**
+ * Given the rate of the clock supplied to the bxCAN macrocell (typically PCLK1) and the desired bit rate,
+ * this function iteratively solves for the best possible timing settings. The CAN bus timing parameters,
+ * such as the sample point location, the number of time quantas per bit, etc., are optimized according to the
+ * recommendations provided in the specifications of UAVCAN, DeviceNet, and CANOpen.
+ *
+ * The implementation is adapted from libuavcan.
+ *
+ * This function is defined in the header in order to encourage the linker to discard it if it is not used.
+ *
+ * The following code has been used to test it (gcc -std=c99 test.c && ./a.out):
+ *
+ *     static void runOnce(const uint32_t pclk1,
```

Comment:
Is there a way to put this code in a unit test instead of in a comment? It would make sure this example code always stays true.